sync alpaka_TESTING to cmake built-in BUILD_TESTING

mehmetyusufoglu opened the pull request:
Problem: BUILD_TESTING and alpaka_TESTING were both visible in ccmake since BUILD_TESTING is cmake build in. But alpaka_TESTING was the correct cmake cache variable to set.
Solution (To be discussed):
`alpaka_TESTING` cache variable is synched to cmake build in `BUILD_TESTING`. It is also converted to a internal variable. 

Only BUILD_TESTING will be visible and settable from command line/cmake.

mehmetyusufoglu committed:
sync alpaka_TESTING to cmake built-in BUILD_TESTING

psychocoderHPC commented:
In principle I refactored the behaviour in https://github.com/alpaka-group/alpaka3/pull/83 as soon as `alpaka_TESTING` is enabled `ctests` worked. CTest currently not works if only `alpaka_EXAMPLES` or `alpaka_BENCHMARKS` is set.

I think only `enable_testing()` must be called in CMake. Let us discuss this in person.

mehmetyusufoglu commented:
Ol, thanks.

René Widera ***@***.***> schrieb am Di., 25. März 2025,
8:57 AM:

> In principle I refactored the behaviour in #83
> <https://github.com/alpaka-group/alpaka3/pull/83> as soon as
> alpaka_TESTING is enabled ctests worked. CTest currently not works if
> only alpaka_EXAMPLES or alpaka_BENCHMARKS is set.
>
> I think only enable_testing() must be called in CMake. Let us discuss
> this in person.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/alpaka-group/alpaka3/pull/94#issuecomment-2750392407>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUYXIJMTGBAHZ3ITJXFRTYD2WD44VAVCNFSM6AAAAABZVRKWU2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONJQGM4TENBQG4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
> [image: psychocoderHPC]*psychocoderHPC* left a comment
> (alpaka-group/alpaka3#94)
> <https://github.com/alpaka-group/alpaka3/pull/94#issuecomment-2750392407>
>
> In principle I refactored the behaviour in #83
> <https://github.com/alpaka-group/alpaka3/pull/83> as soon as
> alpaka_TESTING is enabled ctests worked. CTest currently not works if
> only alpaka_EXAMPLES or alpaka_BENCHMARKS is set.
>
> I think only enable_testing() must be called in CMake. Let us discuss
> this in person.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/alpaka-group/alpaka3/pull/94#issuecomment-2750392407>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUYXIJMTGBAHZ3ITJXFRTYD2WD44VAVCNFSM6AAAAABZVRKWU2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONJQGM4TENBQG4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

mehmetyusufoglu commented:
Solved by another PR
